sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -539,3 +541,90 @@ case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: E
 
   override def prettyName: String = "str_to_map"
 }
+
+/**
+ * Adds/replaces field in struct by name.
+ */
+case class WithField(structExpr: Expression, fieldName: String, valueExpr: Expression)
+  extends Unevaluable {
+
+  private lazy val fieldPath = CatalystSqlParser.parseMultipartIdentifier(fieldName)
```

Comment:
`Expression` is a very low-level unit of SQL execution. It will be serialized and sent to executors, and be evaluated for each task.

I think it makes more sense to parse the field name at the driver side, so it's only done once.

Reply:
Thanks for explaining this to me. I've changed the code back to the way you wrote it in your refactor commit.